### PR TITLE
Move MessageUIStates inside MessageState

### PR DESCRIPTION
**Pull request by fpersico**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/messages/MessageDetailScreen.tsx [hunk 1]
```diff
@@ -152,9 +151,12 @@ export class MessageDetailScreen extends React.PureComponent<Props, never> {
   };
 
   private setMessageReadState = () => {
-    const { potMessage, messageUIStates } = this.props;
+    const { potMessage, maybeUIStates } = this.props;
 
-    if (pot.isSome(potMessage) && !messageUIStates.read) {
+    if (
+      pot.isSome(potMessage) &&
+      maybeUIStates.fold(false, uiStates => !uiStates.read)
```
Comment (cloudify) [inline ts/screens/messages/MessageDetailScreen.tsx]: no very clear the intended behavior here, maybe `getOrElse` would be easier to understand?

File: ts/store/reducers/entities/messages/messagesById.ts [hunk 2]
```diff
@@ -9,13 +9,18 @@ import { getType } from "typesafe-actions";
...
 import { Action } from "../../../actions/types";
 import { GlobalState } from "../../types";
 
+export type MessageUIStates = {
+  read: boolean;
+};
+
 export type MessageState = {
   meta: CreatedMessageWithoutContent;
+  uiStates: MessageUIStates;
```
[the remark this answers is not in the document]
Comment (fpersico) [reply ts/store/reducers/entities/messages/messagesById.ts]: "isArchived" is coming soon :)

File: ts/screens/messages/MessageDetailScreen.tsx (hunk 1 above)
Comment (fpersico) [reply ts/screens/messages/MessageDetailScreen.tsx]: It is the equivalent of
```
    if (
      pot.isSome(potMessage) &&
      maybeUIStates.isSome() &&
      !maybeUIStates.value.read
    ) 

is this form clearer?
```

Comment (fpersico) [reply ts/screens/messages/MessageDetailScreen.tsx]: Changed as above

**Comment by cloudify** [comment]
Still, I can't see any value in nesting these attributes in an additional object

**Comment by cloudify** [comment]
Perhaps maybeUIStates.map(_ => _.read).getOrElse(false)

**Pull request merged**